cpp/src/arrow/flight/sql/server.cc

```diff
@@ -577,6 +579,7 @@ arrow::Result<std::string> CreateStatementQueryTicket(
 Status FlightSqlServerBase::GetFlightInfo(const ServerCallContext& context,
                                           const FlightDescriptor& request,
                                           std::unique_ptr<FlightInfo>* info) {
+  ARROW_FLIGHT_OTELLOG_SQL_SERVER(INFO, "[Example message] func=", __func__);
```

Comment:
They're only here temporarily for demonstration purposes (mainly with the FlightSQL test app/server). I just figured it would be easier to include them alongside everything else before eventually removing them.